### PR TITLE
Improve type definitions.

### PR DESCRIPTION
**Pull request by tripodsgames**
Add support to generics on `Request.json` and `Request.urlencoded`.
Fixed `path_parameters` and `query_parameters` to return `Record<string, unknown>` instead of `Object`.
Change `Object` to `any` on Response.json and Response.jsonp params.

### CODE REVIEW

File: types/components/http/Request.d.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export default class Request {
 
     /**
      * Returns request headers from incoming request.
-     * @returns {Object}
+     * @returns {Record<string, unknown>}
```
Comment (kartikk221) [inline types/components/http/Request.d.ts:149]: This should be Record<string, string> as header keys/values can and will only ever be strings.

File: types/components/http/Request.d.ts [hunk 2]
```diff
@@ -160,13 +160,13 @@ export default class Request {
      * Returns path parameters from incoming request in Object form {key: value}
      * @returns {Record<string, unknown>}
      */
-    get path_parameters(): Object;
+    get path_parameters(): Record<string, unknown>;
```
Comment (kartikk221) [inline types/components/http/Request.d.ts:163]: This should be Record<string, string> as path parameter keys/values can and will only ever be strings.

File: types/components/http/Request.d.ts [hunk 3]
```diff
@@ -160,13 +160,13 @@ export default class Request {
...
 
     /**
      * Returns query parameters from incoming request in Object form {key: value}
      * @returns {Record<string, unknown>}
      */
-    get query_parameters(): Object;
+    get query_parameters(): Record<string, unknown>;
```
Comment (kartikk221) [inline types/components/http/Request.d.ts:169]: This record may be a bit ambiguous as we use the query-string package under the hood to parse query parameters and currently only the default options are used which ensures that only a string value is possible. But in the next update I am planning on adding the ability to configure the options passed to the query parameter parser through Server/Route options.

File: types/components/http/Request.d.ts (hunk 1 above)
Comment (tripodsgames) [reply types/components/http/Request.d.ts]: fixed

File: types/components/http/Request.d.ts (hunk 2 above)
Comment (tripodsgames) [reply types/components/http/Request.d.ts]: fixed

File: types/components/http/Request.d.ts (hunk 3 above)
Comment (tripodsgames) [reply types/components/http/Request.d.ts:169]: I think for now we can keep the `unknown`, what do you think?

Comment (tripodsgames) [reply types/components/http/Request.d.ts:169]: But, if you want, I can change it to `string` and in the next update we can change it to use generics `<T>`.

**Comment by tripodsgames** [comment]
@kartikk221 can you review it again?

**Pull request merged**